### PR TITLE
Add `@MustCallAlias` annotations

### PR DESCRIPTION
**Pull request by mernst**

### CODE REVIEW

File: src/java.base/share/classes/java/util/Arrays.java [hunk 1]
```diff
@@ -5149,7 +5151,7 @@ else if (e1 instanceof boolean[] && e2 instanceof boolean[])
      * @since 1.5
      */
     @SideEffectFree
-    public static @MinLen(2) String toString(@PolyNull @PolyInterned Object @Nullable [] a) {
+    public static @MinLen(2) String toString(@PolyMustCall @PolyNull @PolyInterned Object @PolyMustCall @Nullable [] a) {
```
Comment (kelloggm) [inline src/java.base/share/classes/java/util/Arrays.java:5158]: This `@PolyMustCall` doesn't make sense to me, for two reasons:
1. `toString` itself shouldn't care what `MustCall` types its arguments have
2. even if, for some reason, we do want to annotate this argument to `toString`, `@PolyMustCall` on both the component type and on the array itself seems wrong: the array itself won't have a must-call obligation (at least, as the MustCall type system is used by the RLC), and even if it does I don't see a reason why we need to LUB its type with the MustCall types of the array components. Can you explain the reason for these annotations?

File: src/java.base/share/classes/java/util/zip/ZipFile.java [hunk 2]
```diff
@@ -377,7 +379,11 @@ private ZipEntry getEntry(String name, Function<String, ? extends ZipEntry> func
      * @throws IOException if an I/O error has occurred
      * @throws IllegalStateException if the zip file has been closed
      */
-    public @Nullable InputStream getInputStream(ZipEntry entry) throws IOException {
+    @CFComment({"These @MustCallAlias annotations might not be right.  The",
+      "Javadoc documentation above is not clear.  It seems that closing the",
+      "ZipEntry does close the InputStream, but it is not clear that closing",
+      "the InputStream also closes the ZipEntry."})
```
Comment (kelloggm) [inline src/java.base/share/classes/java/util/zip/ZipFile.java:385]: I agree this is unclear. A brief web search didn't turn up any better information, so I think for now assuming that this method has a standard contract and that the documentation is a bit unclear is the right thing to do. Thanks for the comment that explains your reasoning.

**Review by kelloggm — CHANGES_REQUESTED** [review]
Tests are failing, I think because of some kind of configuration problem. I only looked at the added annotations, not at the changes to the CI setup or the JTREG tests.

**Comment by mernst** [comment]
@kelloggm Can you take another look?  I think I will have the build problems corrected (by pulling master, so there are no such changes here) shortly.

**Review by kelloggm — APPROVED** [review]
Annotations LGTM

**Pull request closed without merging**